### PR TITLE
Remove reaction

### LINKED ISSUE #765: Remove Reaction
### Feature Request

There is `addReaction()`, but no `removeReaction()`. There's no way to automate removing a reaction which is needed for temporary reactions like `:loading`

- [ ] Add feature to remove an added reaction

### PR DESCRIPTION
**Pull request by philippart-s**
Fix #765 

Add the _remove reaction_ feature.

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by philippart-s** [comment]
@timja ok for review
I'don"t factorize code for addReaction and removeReaction to break less things.

File: src/main/java/jenkins/plugins/slack/StandardSlackService.java [hunk 1]
```diff
@@ -427,6 +427,21 @@ public boolean addReaction(String channelId, String timestamp, String emojiName)
...
+    public boolean removeReaction(String channelId, String timestamp, String emojiName) {
+        JSONObject json = SlackReactionRequest.builder()
+                .withChannelId(channelId)
+                .withTimestamp(timestamp)
+                .withEmojiName(emojiName)
+                .build()
+                .getBody();
+
+        logger.fine("Removing reaction:  " + json.toString());
+        return postToSlack("reactions.remove", json);    }
```
Comment (timja) [inline src/main/java/jenkins/plugins/slack/StandardSlackService.java:443]: formatting is off here

File: src/main/java/jenkins/plugins/slack/workflow/SlackResponse.java [hunk 2]
```diff
@@ -65,4 +65,20 @@ public boolean addReaction(String emojiName) {
...
+     *
+     * @return boolean indicating whether the API request succeeded
+     */
+    @Whitelisted
+    public boolean removeReaction(String emojiName) {
+        if (slackService == null) {
+            return false;
+        }
+
+        return slackService.removeReaction(channelId, ts, emojiName);
```
Comment (timja) [inline src/main/java/jenkins/plugins/slack/workflow/SlackResponse.java:82]: formatting is off

File: src/main/java/jenkins/plugins/slack/StandardSlackService.java (hunk 1 above)
Comment (philippart-s) [reply src/main/java/jenkins/plugins/slack/StandardSlackService.java]: fix with https://github.com/jenkinsci/slack-plugin/pull/769/commits/b7f18759113465f2858e7f6f44a661950d615f3e

File: src/main/java/jenkins/plugins/slack/workflow/SlackResponse.java (hunk 2 above)
Comment (philippart-s) [reply src/main/java/jenkins/plugins/slack/workflow/SlackResponse.java:82]: Sorry but I don't saw where formating is broken ?
If I format code with my IDE (Intellij) severals change will be made, is there an editorconfig (or similar) or like to share options of format code ?

File: src/main/java/jenkins/plugins/slack/StandardSlackService.java [hunk 3]
```diff
@@ -427,6 +427,22 @@ public boolean addReaction(String channelId, String timestamp, String emojiName)
         return postToSlack("reactions.add", json);
     }
 
+    /**
+     * Implementation of {@link SlackService#removeReaction(String, String, String)}
```
Comment (timja) [inline src/main/java/jenkins/plugins/slack/StandardSlackService.java:431]:
```suggestion
     * Remove an emoji reaction from a message.
```

**Comment by philippart-s** [comment]
@timja ok for review

File: src/main/java/jenkins/plugins/slack/workflow/SlackResponse.java (hunk 2 above)
Comment (eduardrudko) [reply src/main/java/jenkins/plugins/slack/workflow/SlackResponse.java:82]: Passing by your PR i see no problems with your formatting, @timja can we go with it?

**Comment by patrickpichler** [comment]
Is there any chance that this is getting merged? :slightly_smiling_face:

**Comment by timja** [comment]
> Is there any chance that this is getting merged? 🙂

any chance you can quickly test it for me and see if it meets your needs? I'll merge if so

**Comment by patrickpichler** [comment]
Remove reaction seems to be working :slightly_smiling_face:

**Comment by timja** [comment]
Thanks should be released in the next 20 mins or so

**Pull request merged**